File: rs/minsk/src/analysis/syntax/kind.rs
use strum::EnumIter;

#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, EnumIter)]
pub enum SyntaxKind {
    BadToken,
    EndOfFileToken,

    WhitespaceToken,
    IdentifierToken,
    NumberToken,

    PlusToken,
    MinusToken,
    StarToken,
    SlashToken,
    BangToken,
    AmpersandAmpersandToken,
    PipePipeToken,
    BangEqualsToken,
    EqualsEqualsToken,
    EqualsToken,
    OpenParenthesisToken,
    CloseParenthesisToken,

    TrueKeyword,
    FalseKeyword,

    CompilationUnit,

    AssignmentExpression,
    BinaryExpression,
    LiteralExpression,
    NameExpression,
    UnaryExpression,
    ParenthesizedExpression,
}

File: py/minsk/analysis/syntax/kind.py
from enum import Enum, auto


class SyntaxKind(Enum):
    BadToken = auto()
    EndOfFileToken = auto()

    WhitespaceToken = auto()
    IdentifierToken = auto()
    NumberToken = auto()

    PlusToken = auto()
    MinusToken = auto()
    StarToken = auto()
    SlashToken = auto()
    BangToken = auto()
    AmpersandAmpersandToken = auto()
    PipePipeToken = auto()
    BangEqualsToken = auto()
    EqualsEqualsToken = auto()
    EqualsToken = auto()
    OpenParenthesisToken = auto()
    CloseParenthesisToken = auto()
    OpenBraceToken = auto()
    CloseBraceToken = auto()

    TrueKeyword = auto()
    FalseKeyword = auto()

    CompilationUnit = auto()

    BlockStatement = auto()
    ExpressionStatement = auto()

    AssignmentExpression = auto()
    BinaryExpression = auto()
    LiteralExpression = auto()
    NameExpression = auto()
    ParenthesizedExpression = auto()
    UnaryExpression = auto()
